### PR TITLE
global: migration to invenio-records-rest

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: examples/app.py [hunk 1]
```diff
@@ -130,44 +142,41 @@ def records():
...
+                rec_pid = str(index)
+                # create PID
+                PersistentIdentifier.create(
+                    'recid', rec_pid, object_type='rec', object_uuid=rec_uuid,
+                    status=PIDStatus.REGISTERED
+                )
+                # do translate
+                record = marc21.do(create_record(data))
+                # update with control number
+                record.update({'control_number': rec_pid})
```
Comment (jirikuncar) [inline examples/app.py]: you should be able to use `minter`(https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/minters.py#L32)

Comment (drjova) [inline examples/app.py]: :+1:

File: examples/app.py [hunk 2]
```diff
@@ -85,7 +88,7 @@
     CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
     CELERY_RESULT_BACKEND="cache",
     DEBUG=True,
-    SEARCH_UI_SEARCH_API='invenio_search_ui.api',
+    SEARCH_UI_SEARCH_API='invenio_records_rest.recid_list',
```
Comment (lnielsen) [inline examples/app.py]: This won't work I'm quite sure. ``invenio_records_rest.recid_list`` exists in the API Flask application, hence a ``url_for`` running in the UI Flask application won't be able to generate URLs for the API application.

Comment (drjova) [inline examples/app.py]: I'm not sure what do you mean (please correct me if I'm wrong), but in the example app works fine. https://github.com/inveniosoftware/invenio-search-ui/blob/master/invenio_search_ui/templates/invenio_search_ui/search.html#L44

Comment (jirikuncar) [inline examples/app.py]: @drjova `invenio-search-ui` is by default registered on different Flask instance.

- `app` - "standard application" (`/login`, `/search`, ...)
- `api_app` - contains all REST-API endpoint with custom exception handling (`/api/...`)

File: examples/app.py [hunk 3]
```diff
@@ -103,6 +106,11 @@
 InvenioRecords(app)
 InvenioSearch(app)
 InvenioSearchUI(app)
+InvenioPIDStore(app)
+InvenioRecordsREST(app)
```
Comment (jirikuncar) [inline examples/app.py]: The example works because you register REST endpoint on "default" application object.

File: examples/app.py (hunk 2 above)
Comment (jirikuncar) [inline examples/app.py]: @lnielsen wouldn't be a solution to add custom handler for missing urls that tries to generate the url with other application object?

Comment (lnielsen) [inline examples/app.py]: @jirikuncar Then the two apps needs to know about each other and run in the same process. I.e. you won't be able to run the UI app on set of servers and the API app on another set of servers.

I think it's better to just point to a URL endpoint. Then you can support both e.g. local (``/api/records/``) and remote (``https://api.example.org/records/``) endpoints.

Comment (jirikuncar) [inline examples/app.py]: @lnielsen I agree that explicit (external) URL is better.

**Comment by jirikuncar** [comment]
superseded by #7

**Pull request closed without merging**